### PR TITLE
Add to the Product Button block the support for the Product Query block

### LINKED ISSUE #6785: Product Query: [Atomic Block] Create `Product Image` block
Create a `Product Image block` for inclusion in the new `Product Query block`. It is important that:
- on the editor side, the block has to be client-side rendered.
- on the frontend side, the block has to be server-side rendered with hydration support.
- keep the block independent by the `Query Loop block`, in this way we can use the new block also for future blocks, for example, the `Single Product` block.

### PR DESCRIPTION
**Pull request by gigitux**
This PR is blocked by #6911 

This PR adds to the Product Button block the support for the Product Query block.
Now, on the client side, when the Product Button Block is used in the Product Query block, the markup will be server side generated. Also, no javascript related to the Product Button block will be loaded. 

Fixes #6785

### Testing

N/A

#### User Facing Testing

1. Create a post/page and add the Product Query block.
2. Save it.
3. Open the frontend side and be sure that:
- no JS related to the Product Button block is loaded (the block is rendered on the server).
- The UI of the block is identical to on the editor side.
- The settings are reflected correctly (eg: custom style)
4. Go to WooCommerce Settings > Products > General.
5. Toggle `Redirect to the cart page after successful addition`.
6. Toggle `Enable AJAX add to cart buttons on archives`.
7. Be sure that the Product Button Block is used in the Product Query Block follows these settings.

Also, consider that this block has a different behavior when it is used in the Product Query block as respect when it is used in All Products. We will address this issue in the next iteration #6949

* [x] Do not include in the Testing Notes 

### WooCommerce Visibility

* [ ] WooCommerce Core
* [ ] Feature plugin
* [x] Experimental

### CODE REVIEW

**Review by danielwrobert — APPROVED** [review]
LGTM! 🚀 

I confirmed the following:

- On `trunk`, the All Products block to loads the `product-button-frontend.js` on the frontend.
- On `trunk`, if I add a Product Button block to the Product Query block, the script is also loaded. Note that the overall behavior is a little odd in this particular scenario but I imagine that to be expected.
- On this PR branch, adding a Product Button block to the PQ block does _not_ render the script on the frontend and the behavior / UI is as expected.
- Confirmed the **Add to cart behaviour** in **WooCommerce > Settings** behaves as expected.

Nice work! 🍻

**Comment by gigitux** [comment]
Thanks for the review! @danielwrobert 🚀

**Comment by sunyatasattva** [comment]
Just leaving this for posterity: this PR actually fixes https://github.com/woocommerce/woocommerce-blocks/issues/6794 not  #6785

File: assets/js/atomic/blocks/product-elements/button/edit.js [hunk 1]
```diff
@@ -1,33 +1,30 @@
 /**
  * External dependencies
  */
-import { __ } from '@wordpress/i18n';
 import { Disabled } from '@wordpress/components';
 import { useBlockProps } from '@wordpress/block-editor';
+import { useEffect } from 'react';
```
Comment (dinhtungdu) [inline assets/js/atomic/blocks/product-elements/button/edit.js:6]: Is there any reason why we don't use `useEffect` from `@wordpress/element` here?

_(Came across this PR while exploring the Query Block.)_

Comment (gigitux) [reply assets/js/atomic/blocks/product-elements/button/edit.js:6]: Absolutely no :'(  
During the cooldown, I guess that it makes sense to work on this https://github.com/woocommerce/woocommerce-blocks/issues/6732

I created a PR and assigned you as a reviewer https://github.com/woocommerce/woocommerce-blocks/pull/7298. Thanks for catching this!

**Pull request merged**